sdk/core/Azure.Core/tests/BearerTokenAuthenticationPolicyTests.cs

```diff
@@ -789,6 +789,109 @@ public async Task BearerTokenAuthenticationPolicy_BackgroundRefreshFailsAndLogs(
             Assert.IsTrue(logged);
         }
 
+        [Test]
+        public async Task BearerTokenAuthenticationPolicy_SwitchedTenants()
```

Comment:
This feels like it should be a test for the real `ChallengeBasedAuthenticationPolicy`. The unit test here could be simplified to just validate that the cache recognizes a change in tenantId.

Reply:
I already added one in #35474 but you said you wanted one local to Azure.Core.

Reply:
Ah, right. But I think the local one here should only need to validate the new token cache invalidation behavior.

Reply:
It should now with the latest push. Please have another look.